doc: fixed inconsistencies in documentation between autotools to cmake change

kevkevinpal opened the pull request:
A bit of a followup from https://github.com/bitcoin/bitcoin/pull/30840

- In this change the documentation where we refer to the `./configure` script which is now gone and have converted the configure params to use the `cmake` equivalent.

kevkevinpal committed:
doc: Updating docs from autotools to cmake

replaced --enable-debug with -DCMAKE_BUILD_TYPE=Debug in developer-notes
replaced --enable-multiprocess with -DWITH_MULTIPROCESS=ON
replaced --disable-zmq with -DWITH_ZMQ=OFF

pablomartin4btc reviewed (commented):
ACK dd0941cc6de6852c8ee1347c7ccfcc887cd07e4b

nit: since you are there at `doc/zmq.md`, I think there's a typo close to the end, look for "chain--as".

fanquake commented on doc/multiprocess.md:
````diff
@@ -23,9 +23,9 @@ src/bitcoin-node -regtest -printtoconsole -debug=ipc
 BITCOIND=bitcoin-node test/functional/test_runner.py
 ```
 
-The configure script will pick up settings and library locations from the depends directory, so there is no need to pass `--enable-multiprocess` as a separate flag when using the depends system (it's controlled by the `MULTIPROCESS=1` option).
````
A few lines above here. This
```
cd <BITCOIN_SOURCE_DIRECTORY>
make -C depends NO_QT=1 MULTIPROCESS=1
CONFIG_SITE=$PWD/depends/x86_64-pc-linux-gnu/share/config.site ./configure
make
src/bitcoin-node -regtest -printtoconsole -debug=ipc
BITCOIND=bitcoin-node test/functional/test_runner.py
```

should be replaced with:
```bash
cd <BITCOIN_SOURCE_DIRECTORY>
make -C depends NO_QT=1 MULTIPROCESS=1
cmake -B build --toolchain=depends/x86_64-pc-linux-gnu/toolchain.cmake
cmake --build build
build/src/bitcoin-node -regtest -printtoconsole -debug=ipc
BITCOIND=build/src/bitcoin-node build/test/functional/test_runner.py
```

Although, it also looks like that is currently broken:
```bash
BITCOIND=build/src/bitcoin-node build/test/functional/test_runner.py
Temporary test directory at /tmp/test_runner_₿_🏃_20240913_111920
2024-09-13T11:19:20.313000Z TestFramework (INFO): PRNG seed is: 4821438991961925939
2024-09-13T11:19:20.315000Z TestFramework (INFO): Initializing test directory /tmp/test_runner_₿_🏃_20240913_111920/cache
2024-09-13T11:19:20.316000Z TestFramework (ERROR): Unexpected exception caught during testing
Traceback (most recent call last):
  File "/root/ci_scratch/test/functional/test_framework/test_framework.py", line 131, in main
    self.setup()
  File "/root/ci_scratch/test/functional/test_framework/test_framework.py", line 301, in setup
    self.setup_chain()
  File "/root/ci_scratch/test/functional/test_framework/test_framework.py", line 392, in setup_chain
    self._initialize_chain()
  File "/root/ci_scratch/test/functional/test_framework/test_framework.py", line 853, in _initialize_chain
    self.start_node(CACHE_NODE_ID)
  File "/root/ci_scratch/test/functional/test_framework/test_framework.py", line 553, in start_node
    node.start(*args, **kwargs)
  File "/root/ci_scratch/test/functional/test_framework/test_node.py", line 256, in start
    self.process = subprocess.Popen(self.args + extra_args, env=subp_env, stdout=stdout, stderr=stderr, cwd=cwd, **kwargs)
                   ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/lib/python3.12/subprocess.py", line 1026, in __init__
    self._execute_child(args, executable, preexec_fn, close_fds,
  File "/usr/lib/python3.12/subprocess.py", line 1955, in _execute_child
    raise child_exception_type(errno_num, err_msg, err_filename)
FileNotFoundError: [Errno 2] No such file or directory: 'build/src/bitcoin-node'
2024-09-13T11:19:20.367000Z TestFramework (INFO): Stopping nodes
2024-09-13T11:19:20.367000Z TestFramework (WARNING): Not cleaning up dir /tmp/test_runner_₿_🏃_20240913_111920/cache
2024-09-13T11:19:20.367000Z TestFramework (ERROR): Test failed. Test logging available at /tmp/test_runner_₿_🏃_20240913_111920/cache/test_framework.log
2024-09-13T11:19:20.367000Z TestFramework (ERROR): 
2024-09-13T11:19:20.368000Z TestFramework (ERROR): Hint: Call /root/ci_scratch/test/functional/combine_logs.py '/tmp/test_runner_₿_🏃_20240913_111920/cache' to consolidate all logs
2024-09-13T11:19:20.368000Z TestFramework (ERROR): 
2024-09-13T11:19:20.368000Z TestFramework (ERROR): If this failure happened unexpectedly or intermittently, please file a bug and provide a link or upload of the combined log.
2024-09-13T11:19:20.368000Z TestFramework (ERROR): https://github.com/bitcoin/bitcoin/issues
2024-09-13T11:19:20.368000Z TestFramework (ERROR): 
Traceback (most recent call last):
  File "/root/ci_scratch/build/test/functional/test_runner.py", line 950, in <module>
    main()
  File "/root/ci_scratch/build/test/functional/test_runner.py", line 571, in main
    run_tests(
  File "/root/ci_scratch/build/test/functional/test_runner.py", line 624, in run_tests
    subprocess.check_output([sys.executable, tests_dir + 'create_cache.py'] + flags + ["--tmpdir=%s/cache" % tmpdir])
  File "/usr/lib/python3.12/subprocess.py", line 466, in check_output
    return run(*popenargs, stdout=PIPE, timeout=timeout, check=True,
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/lib/python3.12/subprocess.py", line 571, in run
    raise CalledProcessError(retcode, process.args,
subprocess.CalledProcessError: Command '['/usr/bin/python3', '/root/ci_scratch/build/test/functional/create_cache.py', '--cachedir=/root/ci_scratch/build/test/cache', '--configfile=/root/ci_scratch/build/test/functional/../config.ini', '--tmpdir=/tmp/test_runner_₿_🏃_20240913_111920/cache']' returned non-zero exit status 1.
```

kevkevinpal replied:
thanks for the feedback! 

I updated this to 
```
cd <BITCOIN_SOURCE_DIRECTORY>
make -C depends NO_QT=1 MULTIPROCESS=1
cmake -B build --toolchain=depends/x86_64-pc-linux-gnu/toolchain.cmake
cmake --build build
build/src/bitcoind -regtest -printtoconsole -debug=ipc
BITCOIND=$(pwd)/build/src/bitcoind build/test/functional/test_runner.py
```

Which does start running the tests please let me know if this looks ok

tdb3 replied:
Wouldn't we want to keep `bitcoin-node` (part of multiprocess) rather than change to `bitcoind`? Am I missing something?

Thoughts @ryanofsky ?

ryanofsky replied:
re: https://github.com/bitcoin/bitcoin/pull/30875#discussion_r1759786194

Thanks for the ping! Yes, this should say `bitcoin-node` not `bitcoind` to test the multiprocess binary. Either command line will work, but `bitcoin-node` will make the test more meaningful, especially with changes from #10102 merged, because `bitcoind` will run tests with the node and wallet in the same process, while `bitcoin-node` will spawn separate `bitcoin-wallet` processes

kevkevinpal replied:
I went ahead and changed it back from `bitcoind` to `bitcoin-node` in [dc2e223](https://github.com/bitcoin/bitcoin/pull/30875/commits/dc2e2231036ec09fac4529d54881457fa9a7dc51)

tdb3 reviewed (commented):
Approach ACK

Thanks for the doc cleanup.

kevkevinpal commented on doc/developer-notes.md:
```diff
@@ -1063,7 +1063,7 @@ bool Chainstate::PreciousBlock(BlockValidationState& state, CBlockIndex* pindex)
 
 - Build and run tests with `-DDEBUG_LOCKORDER` to verify that no potential
   deadlocks are introduced. As of 0.12, this is defined by default when
```
thanks! I removed that in [5a85022](https://github.com/bitcoin/bitcoin/pull/30875/commits/5a85022cc6c315e9c756bbbb63a7ed992e7f23d3)

kevkevinpal commented on doc/multiprocess.md:
```diff
@@ -23,9 +23,9 @@ src/bitcoin-node -regtest -printtoconsole -debug=ipc
+The configure script will pick up settings and library locations from the depends directory, so there is no need to pass `-DWITH_MULTIPROCESS=ON` as a separate flag when using the depends system (it's controlled by the `MULTIPROCESS=1` option).
 
-Alternately, you can install [Cap'n Proto](https://capnproto.org/) and [libmultiprocess](https://github.com/chaincodelabs/libmultiprocess) packages on your system, and just run `./configure --enable-multiprocess` without using the depends system. The configure script will be able to locate the installed packages via [pkg-config](https://www.freedesktop.org/wiki/Software/pkg-config/). See [Install ...
+Alternately, you can install [Cap'n Proto](https://capnproto.org/) and [libmultiprocess](https://github.com/chaincodelabs/libmultiprocess) packages on your system, and just run `cmake -B build -DWITH_MULTIPROCESS=ON` without using the depends system. The configure script will be able to locate the installed packages via [pkg-config](https://www.freedesktop.org/wiki/Software/pkg-config/). See [Ins ...
```
thanks! I updated this to `cmake build` instead of `configure script` in  [5a85022](https://github.com/bitcoin/bitcoin/pull/30875/commits/5a85022cc6c315e9c756bbbb63a7ed992e7f23d3)

let me know if you'd prefer different wording!

fanquake commented:
Also in `test_deterministic_coverage.sh`: `Run \"./configure --enable-lcov\"`

jonatack reviewed (commented):
Concept ACK

pablomartin4btc commented:
Since you are here, you could update the translate instructions as well (I've tested them), if @hebasto agrees with the changes:

- `doc/translation_process.md`

   ```diff
   @@ -18,8 +18,8 @@ We use automated scripts to help extract translations in both Qt, and non-Qt sou
    
    To automatically regenerate the `bitcoin_en.ts` file, run the following commands:
    ```sh
   -cd src/
   -make translate
   +cmake -B build -DWITH_BDB=ON -DBUILD_GUI=ON
   +cmake --build build --target translate
    
   **Example Qt translation**
   
   ```
- `doc/translation_strings_policy.md`
   ```diff
   @@ -97,4 +97,4 @@ The second example reduces the number of pluralized words that translators have
    
    During a string freeze (often before a major release), no translation strings are to be added, modified or removed.
    
   -This can be checked by executing `make translate` in the `src` directory, then verifying that `bitcoin_en.ts` remains unchanged.
   +This can be checked by building `translate` package with `cmake` ([instructions](/doc/translate_process.md)), then verifying that `bitcoin_en.ts` remains unchanged.
   
```

kevkevinpal commented on doc/multiprocess.md:
```diff
@@ -4,7 +4,7 @@ _This document describes usage of the multiprocess feature. For design informati
 ## Build Option
 
-On unix systems, the `--enable-multiprocess` build option can be passed to `./configure` to build new `bitcoin-node`, `bitcoin-wallet`, and `bitcoin-gui` executables alongside existing `bitcoind` and `bitcoin-qt` executables.
+On unix systems, the `-DWITH_MULTIPROCESS=ON` build option can be passed to `cmake -B build` to build new `bitcoin-node`, `bitcoin-wallet`, and `bitcoin-gui` executables alongside existing `bitcoind` and `bitcoin-qt` executables.
```
thank you! Updated in [4f72eca](https://github.com/bitcoin/bitcoin/pull/30875/commits/4f72ecae2b542dbc052fae78cdc5b64a0da8b534)

ryanofsky replied:
> @ryanofsky `bitcoin-wallet` is built, if enabled, independently of the multiprocess flag (if I'm not confused); should `doc/multiprocess.md` describe this differently?

Probably previous description was a little confusing but current version c3d68e11191e036e154916f27f21200d2bf16756 which doesn't mention bitcoin-wallet looks good to me.

This readme was written as part of #10102 before bitcoin-wallet tool existed, so --enable-multiprocess option really did build a new binary that didn't exist before. But now #10102 just adds multiprocess functionality to the existing binary. In the master branch currently though multiproces option has no effect on bitcoin-wallet so it is best not to mention it.

kevkevinpal commented on doc/zmq.md:
```diff
@@ -163,7 +163,7 @@ Note that for `*block` topics, when the block chain tip changes,
 It is up to the subscriber to retrieve the chain from the last known
 block to the new tip. Also note that no notification will occur if the tip
-was in the active chain--as would be the case after calling invalidateblock RPC.
+was in the active chain, as would be the case after calling invalidateblock RPC.
```
thank you! Updated in [4f72eca](https://github.com/bitcoin/bitcoin/pull/30875/commits/4f72ecae2b542dbc052fae78cdc5b64a0da8b534)

kevkevinpal commented on src/qt/README.md:
```diff
@@ -99,7 +99,7 @@ sudo apt-get install qtcreator
 
 1. Make sure you've installed all dependencies specified in your systems build instructions
-2. Follow the compile instructions for your system, run `./configure` with the `--enable-debug` flag
+2. Follow the compile instructions for your system, run `cmake -B build` with the `-DCMAKE_BUILD_TYPE=Debug` flag
```
thank you! Updated in [4f72eca](https://github.com/bitcoin/bitcoin/pull/30875/commits/4f72ecae2b542dbc052fae78cdc5b64a0da8b534)

kevkevinpal commented on test/functional/test_runner.py:
```diff
@@ -488,7 +488,7 @@ def main():
     if not enable_bitcoind:
         print("No functional tests to run.")
-        print("Rerun ./configure with --with-daemon and then make")
+        print("Rerun cmake -B build with -DBUILD_DAEMON=ON and then make")
```
thank you! Updated in [4f72eca](https://github.com/bitcoin/bitcoin/pull/30875/commits/4f72ecae2b542dbc052fae78cdc5b64a0da8b534)

kevkevinpal commented:
> Also in `test_deterministic_coverage.sh`: `Run \"./configure --enable-lcov\"`

thank you! Updated in [4f72eca](https://github.com/bitcoin/bitcoin/pull/30875/commits/4f72ecae2b542dbc052fae78cdc5b64a0da8b534)

kevkevinpal commented:
> Since you are here, you could update the translate instructions as well (I've tested them), if @hebasto agrees with the changes:
> 
>     * `doc/translation_process.md`
>       ```diff
>       @@ -18,8 +18,8 @@ We use automated scripts to help extract translations in both Qt, and non-Qt sou
>        
>        To automatically regenerate the `bitcoin_en.ts` file, run the following commands:
>        ```sh
>       -cd src/
>       -make translate
>       +cmake -B build -DWITH_BDB=ON -DBUILD_GUI=ON
>       +cmake --build build --target translate
>        
>       **Example Qt translation**
>       ```
> 
>     * `doc/translation_strings_policy.md`
>       ```diff
>       @@ -97,4 +97,4 @@ The second example reduces the number of pluralized words that translators have
>        
>        During a string freeze (often before a major release), no translation strings are to be added, modified or removed.
>        
>       -This can be checked by executing `make translate` in the `src` directory, then verifying that `bitcoin_en.ts` remains unchanged.
>       +This can be checked by building `translate` package with `cmake` ([instructions](/doc/translate_process.md)), then verifying that `bitcoin_en.ts` remains unchanged.
>       ```

thank you! Updated in [4f72eca](https://github.com/bitcoin/bitcoin/pull/30875/commits/4f72ecae2b542dbc052fae78cdc5b64a0da8b534)

@hebasto if you disagree I can change them back!

kevkevinpal commented:
From these two comments these are the two that have not been addressed yet in this PR 
https://github.com/bitcoin/bitcoin/pull/30664#issuecomment-2320892444 https://github.com/bitcoin/bitcoin/pull/30664#issuecomment-2321610186

```
test/get_previous_releases.py: './configure {}'.format(config_flags),
contrib/devtools/check-deps.sh:# Output makefile targets, converting library .a paths to libtool .la targets
```

hebasto commented:
@kevkevinpal 

I didn't notice that https://github.com/bitcoin/bitcoin/pull/30902 overlaps with this PR. Feel free to pick all changes from https://github.com/bitcoin/bitcoin/pull/30902 to let me close it.

kevkevinpal commented:
> @kevkevinpal
> 
> I didn't notice that #30902 overlaps with this PR. Feel free to pick all changes from #30902 to let me close it.
> 
> UPD. Please separate CI changes into a dedicated commit with the `ci: ` prefix.

We can keep your PR for now and I just removed any overlapping changes in [9e4f9c9](https://github.com/bitcoin/bitcoin/pull/30875/commits/9e4f9c98e30985a5033bcb69d97adcd7f858adb1)

fanquake commented on depends/packages.md:
```diff
@@ -152,8 +152,7 @@ Most autotools projects can be properly staged using:
 
 ## Build outputs:
 
-In general, the output of a depends package should not contain any libtool
```
This line can remain: https://github.com/bitcoin/bitcoin/pull/30902#discussion_r1759727845.

kevkevinpal replied:
no problem removed in [03fd466](https://github.com/bitcoin/bitcoin/pull/30875/commits/03fd466581666cfd637e79a8594a9a0e7d097524)[03fd466](https://github.com/bitcoin/bitcoin/pull/30875/commits/03fd466581666cfd637e79a8594a9a0e7d097524)[03fd466](https://github.com/bitcoin/bitcoin/pull/30875/commits/03fd466581666cfd637e79a8594a9a0e7d097524)

fanquake commented:
https://github.com/bitcoin/bitcoin/blob/37679b856ce183ec256c12a680b93ad53ed94da6/doc/design/libraries.md?plain=1#L11

kevkevinpal commented:
> https://github.com/bitcoin/bitcoin/blob/37679b856ce183ec256c12a680b93ad53ed94da6/doc/design/libraries.md?plain=1#L11

Thanks! updated in [c3d68e1](https://github.com/bitcoin/bitcoin/pull/30875/commits/c3d68e11191e036e154916f27f21200d2bf16756)

davidgumberg commented:
Concept ACK

Nit: could also fix

```diff
--- a/test/util/test_runner.py
+++ b/test/util/test_runner.py
@@ -5,7 +5,7 @@
 # file COPYING or http://www.opensource.org/licenses/mit-license.php.
 """Test framework for bitcoin utils.

-Runs automatically during `make check`.
+Runs automatically during `ctest --test-dir build/`.

 Can also be run manually."""
 ```

kevkevinpal commented on doc/developer-notes.md:
```diff
@@ -429,9 +429,9 @@ Defining `DEBUG_LOCKCONTENTION` adds a "lock" logging category to the logging
+The `-DCMAKE_BUILD_TYPE=Debug` build option adds `-DDEBUG_LOCKCONTENTION` to the
 compiler flags. You may also enable it manually for a non-debug build by running
-configure with `-DDEBUG_LOCKCONTENTION` added to your CPPFLAGS,
+build with `-DDEBUG_LOCKCONTENTION` added to your CPPFLAGS,
```
thank you! Addressed this in [bc4a929](https://github.com/bitcoin/bitcoin/pull/30875/commits/bc4a929cd716cd2b412c70754749d4fda0ca2a10)

kevkevinpal commented on doc/translation_strings_policy.md:
```diff
@@ -97,4 +93,4 @@ The second example reduces the number of pluralized words that translators have
 During a string freeze (often before a major release), no translation strings are to be added, modified or removed.
 
-This can be checked by executing `make translate` in the `src` directory, then verifying that `bitcoin_en.ts` remains unchanged.
+This can be checked by building `translate` package with `cmake` ([instructions](translation_process.md)), then verifying that `bitcoin_en.ts` remains unchanged.
```
thank you! Addressed this in [bc4a929](https://github.com/bitcoin/bitcoin/pull/30875/commits/bc4a929cd716cd2b412c70754749d4fda0ca2a10)

I will take a look at the backtick nit, to see if I can find already existing instances of `cmake` with no backticks

kevkevinpal commented on test/functional/test_runner.py:
```diff
@@ -488,7 +488,7 @@ def main():
     if not enable_bitcoind:
         print("No functional tests to run.")
-        print("Rerun ./configure with --with-daemon and then make")
+        print("Re-compile  with the -DBUILD_DAEMON=ON build option")
```
thank you! Addressed this in [bc4a929](https://github.com/bitcoin/bitcoin/pull/30875/commits/bc4a929cd716cd2b412c70754749d4fda0ca2a10)

kevkevinpal commented:
> Concept ACK
> 
> Nit: could also fix
> 
> ```diff
> --- a/test/util/test_runner.py
> +++ b/test/util/test_runner.py
> @@ -5,7 +5,7 @@
>  # file COPYING or http://www.opensource.org/licenses/mit-license.php.
>  """Test framework for bitcoin utils.
> 
> -Runs automatically during `make check`.
> +Runs automatically during `ctest --test-dir build/`.
> 
>  Can also be run manually."""
> ```

thank you! Addressed this in [bc4a929](https://github.com/bitcoin/bitcoin/pull/30875/commits/bc4a929cd716cd2b412c70754749d4fda0ca2a10)

davidgumberg commented:
ACK https://github.com/bitcoin/bitcoin/commit/bc4a929cd716cd2b412c70754749d4fda0ca2a10

pico-nit: your branch messes up the 80 character wrapping in a few spots in `doc/developer-notes.md`. This is not really consistent through the docs anyways, so I don't mind personally.

kevkevinpal commented on src/util/fs_helpers.cpp:
```diff
@@ -22,7 +22,7 @@
 
 #ifndef WIN32
-// for posix_fallocate, in configure.ac we check if it is present after this
+// for posix_fallocate, in cmake -B build we check if it is present after this
```
thanks! Updated in [a9964c0](https://github.com/bitcoin/bitcoin/pull/30875/commits/a9964c04447745435747d9cc557165c43902783b)

kevkevinpal commented on doc/translation_strings_policy.md:
```diff
@@ -97,4 +93,4 @@ The second example reduces the number of pluralized words that translators have
 During a string freeze (often before a major release), no translation strings are to be added, modified or removed.
 
-This can be checked by executing `make translate` in the `src` directory, then verifying that `bitcoin_en.ts` remains unchanged.
+This can be checked by building the `translate` package with `cmake` ([instructions](translation_process.md)), then verifying that `bitcoin_en.ts` remains unchanged.
```
thanks! Updated in [a9964c0](https://github.com/bitcoin/bitcoin/pull/30875/commits/a9964c04447745435747d9cc557165c43902783b)

kevkevinpal commented on doc/translation_process.md:
````diff
@@ -18,8 +18,8 @@ We use automated scripts to help extract translations in both Qt, and non-Qt sou
 ```sh
-cd src/
-make translate
+cmake -B build -DWITH_BDB=ON -DBUILD_GUI=ON
````
thanks! Updated in [a9964c0](https://github.com/bitcoin/bitcoin/pull/30875/commits/a9964c04447745435747d9cc557165c43902783b)

kevkevinpal commented:
> ACK [bc4a929](https://github.com/bitcoin/bitcoin/commit/bc4a929cd716cd2b412c70754749d4fda0ca2a10)
> 
> pico-nit: your branch messes up the 80 column wrapping in a few spots in `doc/developer-notes.md`. This is not really consistent through the docs anyways, so I don't mind personally.

thanks for the suggestion! But I think I'll leave those for now since they don't relate to the autotools to cmake update too much and I would prefer to keep the PR more focused on those changes

jarolrod reviewed (approved):
ACK a9964c04447745435747d9cc557165c43902783b

Tried to find other inconsistencies, but couldn't.

tdb3 reviewed (approved):
ACK a9964c04447745435747d9cc557165c43902783b

pablomartin4btc reviewed (commented):
re-ACK a9964c04447745435747d9cc557165c43902783b
